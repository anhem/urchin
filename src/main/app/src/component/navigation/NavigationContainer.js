import React, {Component} from 'react'
import {connect} from 'react-redux'
import {toggleNavigation} from '../../action/siteAction';
import Navigation from './Navigation';

export class NavigationContainer extends Component {

    render() {
        return (
            <Navigation navigationVisible={this.props.navigationVisible}
                        callbacks={{
                            toggleNavigation: this.props.toggleNavigation,
                        }}/>
        )
    }
}

const mapDispatchToProps = (dispatch) => {
    return {
        toggleNavigation: (navigationVisible) => {
            dispatch(toggleNavigation(navigationVisible))
        }
    }
};

const mapStateToProps = (state) => {
    return {
        navigationVisible: state.siteData.navigationVisible || false
    }
};

export default connect(mapStateToProps, mapDispatchToProps)(NavigationContainer)