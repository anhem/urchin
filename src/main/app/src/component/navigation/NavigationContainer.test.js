import React from 'react';
import {mount, shallow} from 'enzyme'
import toJson from 'enzyme-to-json';
import {NavigationContainer} from "./NavigationContainer";

describe('NewGroupContainer', () => {

    it('renders without crashing', () => {

        expect(mount(<NavigationContainer/>).length).toEqual(1);
    });

    it('match snapshot', () => {
        expect(toJson(shallow(<NavigationContainer/>))).toMatchSnapshot();
    });
});
