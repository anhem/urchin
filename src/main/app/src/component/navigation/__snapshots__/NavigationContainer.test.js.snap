// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewGroupContainer match snapshot 1`] = `
<Navigation
  callbacks={
    Object {
      "toggleNavigation": undefined,
    }
  }
/>
`;
